span.go

```diff
@@ -47,6 +35,46 @@ type Span interface {
 	// Suitable for serializing over the wire, etc.
 	TraceContext() TraceContext
 
+	// Log() records `data` to this Span at `timestamp`.
+	//
+	// See LogData for semantic details.
+	Log(timestamp time.Time, data LogData)
+
+	// Info() is equivalent to
+	//
+	//   Log(time.Now(), LogData{Message: fmt.Sprint(args...)})
+	//
+	Info(args ...interface{})
+
+	// Infof() is equivalent to
+	//
+	//   Log(time.Now(), LogData{Message: fmt.Sprintf(format, args...)})
```

Comment:
Yes, the idea here (per option A) was to restrict application payload data (like a json, as you say) to either `Event("payload_identifier", appPayload)` or the catch-all `Log(time.Now(), ...)`.

I have been looking at code we have where application payloads are involved, and for the most part there's either a pretty-printed log message with format params XOR a payload; rarely both.